export/securedrop_export/print/service.py

```diff
@@ -210,53 +235,150 @@ def _setup_printer(self, printer_uri, printer_ppd):
 
     def _print_test_page(self):
         logger.info("Printing test page")
-        self._print_file("/usr/share/cups/data/testprint")
+        testprint = Path("/usr/share/cups/data/testprint")
+        self._print_file(testprint)
 
     def _print_all_files(self):
-        files_path = os.path.join(self.submission.tmpdir, "export_data/")
-        files = os.listdir(files_path)
+        print_directory = Path(Path(self.submission.tmpdir, "export_data"))
+        files = os.listdir(print_directory)
         for print_count, f in enumerate(files):
-            file_path = os.path.join(files_path, f)
+            file_path = Path(print_directory, f)
             self._print_file(file_path)
-            logger.info(f"Printing document {print_count} of {len(files)}")
-
-    def _is_open_office_file(self, filename):
-        OPEN_OFFICE_FORMATS = [
-            ".doc",
-            ".docx",
-            ".xls",
-            ".xlsx",
-            ".ppt",
-            ".pptx",
-            ".odt",
-            ".ods",
-            ".odp",
-            ".rtf",
+            logger.info(f"Printing document {print_count + 1} of {len(files)}")
+
+    def _get_supported_mimetypes_libreoffice(self, desktop_dir: Path):
+        """
+        Return a list of mimetypes supported by Libreoffice programs.
+
+        desktop_dir is a path, such as /usr/share/applications/, specified
+        by the constant LIBREOFFICE_DESKTOP_FILES.
+        """
+        supported_mimetypes: set[str] = set()
+        libreoffice_programs = [
+            "libreoffice-base",
+            "libreoffice-calc",
+            "libreoffice-draw",
+            "libreoffice-impress",
+            "libreoffice-math",
+            "libreoffice-writer",
         ]
-        for extension in OPEN_OFFICE_FORMATS:
-            if os.path.basename(filename).endswith(extension):
-                return True
-        return False
+        for item in libreoffice_programs:
+            desktop_file = Path(desktop_dir, f"{item}.desktop")
+            if desktop_file.exists():
+                with open(desktop_file) as f:
```

Comment:
nit: You could avoid some extra indentation with `for line in desktop_file.read_text().splitlines()`.